java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1244,6 +1248,36 @@ public CompletableFuture<String> xadd(
         return commandManager.submitNewCommand(XAdd, arguments, this::handleStringOrNullResponse);
     }
 
+    @Override
+    public CompletableFuture<Map<String, Map<String, String[][]>>> xread(
+            @NonNull Map<String, String> keysAndIds) {
+        return xread(keysAndIds, StreamReadOptions.builder().build());
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Map<String, String[][]>>> xread(
+            @NonNull Map<String, String> keysAndIds, @NonNull StreamReadOptions options) {
+        String[] arguments = options.toArgs(keysAndIds);
+        return commandManager.submitNewCommand(
+                XRead,
+                arguments,
+                response -> {
+                    Map<String, Object> mapResponse = handleMapOrNullResponse(response);
+                    if (mapResponse == null) {
+                        return null;
+                    }
+                    return mapResponse.entrySet().stream()
+                            .collect(
+                                    Collectors.toMap(
+                                            Map.Entry::getKey,
+                                            e -> {
+                                                return castMapOf2DArray(
```

Comment:
@Yury-Fridlyand I added `castMapOf2DArray` but I didn't think a Map of Map of 2D array was very re-usable.  Let me know what you think.

Reply:
Please create function `<Map<String, Map<String, String[][]>> handleXReadResponse(Response)` and call it there on line 1265 `this::handleXReadResponse`.
The code is OK, just move it to another place.